crates/subspace-farmer/src/cluster/controller/farms.rs

```diff
@@ -136,28 +161,32 @@ impl KnownFarmers {
         available_indices
     }
 
-    fn remove_expired(
-        &mut self,
-    ) -> impl Iterator<Item = (ClusterFarmerId, FarmIndex, FarmId)> + '_ {
+    fn check_expired(&mut self) -> impl Iterator<Item = (ClusterFarmerId, &FarmIndex, &FarmId)> {
```

Comment:
This is still `remove_expired()` to me since it removes things, even though not immediately. `check_expired` is a bit misleading, I'd expect for something to be done about returned entries, but it is not necessary.